File: c#/src/WOT.TAK.Connection/src/ConnectorFactory.cs
using Microsoft.Extensions.Options;

namespace WOT.TAK.Connection;

public class ConnectorFactory
{
    private readonly ConnectorSettings _settings;

    public ConnectorFactory(IOptionsSnapshot<ConnectorSettings> settings)
    {
        _settings = settings.Value;
    }

    public TAKServerConnector GetSSLConnector()
    {
        return new SSLConnector(
            _settings.ServerUrl,
            _settings.ServerPort,
            _settings.CotResponsesDirPath);
           // new SocketFactory(_settings));
    }

    public TAKServerConnector GetTAKServerConnector()
    {
        return new CredentialsConnector(
            _settings.ServerUrl,
            _settings.ServerPort,
            _settings.CotResponsesDirPath,
            new SocketFactory(_settings),
            _settings.Login,
            _settings.Password);
    }

    public TAKServerConnector GetPlainConnector()
    {
        return new PlainConnector(
            _settings.ServerUrl,
            _settings.ServerPort,
            _settings.CotResponsesDirPath,
            new SocketFactory(_settings));
    }

    public TAKServerConnector GetUDPConnector()
    {
        return new UDPConnector(
            _settings.ServerUrl,
            _settings.ServerPort);
    }
    public TAKServerConnector GetTCPConnector()
    {
        return new TCPConnector(
            _settings.ServerUrl,
            _settings.ServerPort,
            _settings.CotResponsesDirPath);
         //   new SocketFactory(_settings));
    }
}

File: src/main/java/wot/tak/connection/ConnectorFactory.java
package wot.tak.connection;

public class ConnectorFactory {

    ConnectorFactoryConfig cfg;

    public ConnectorFactory(ConnectorFactoryConfig cfg) {
        this.cfg = cfg;
    }

    public TAKServerConnector getSSLConnector() throws Exception{
        return new SSLConnector(
                cfg.getTakServerUrl(),
                cfg.getTakServerPort(),
                cfg.getCotResponsesPath(),
                new SocketFactory(cfg.getSocketFactoryConfig())
        );
    }

    public TAKServerConnector getSSLCredConnector() throws Exception{
        return new SSLCredConnector(
                cfg.getTakServerUrl(),
                cfg.getTakServerPort(),
                cfg.getCotResponsesPath(),
                new SocketFactory(cfg.getSocketFactoryConfig()),
                cfg.getUserName(),
                cfg.getUserPassword()
        );
    }

    public TAKServerConnector getCredentialsConnector() throws Exception{
        return new CredentialsConnector(
                cfg.getTakServerUrl(),
                cfg.getTakServerPort(),
                cfg.getCotResponsesPath(),
                new SocketFactory(cfg.getSocketFactoryConfig()),
                cfg.getUserName(),
                cfg.getUserPassword()
        );
    }

    public TAKServerConnector getPlainConnector() throws Exception{
        return new PlainConnector(
                cfg.getTakServerUrl(),
                cfg.getTakServerPort(),
                cfg.getCotResponsesPath(),
                new SocketFactory(cfg.getSocketFactoryConfig())
        );
    }

    public TAKServerConnector getUDPConnector() throws Exception{
        return new UDPConnector(
                cfg.getUserName(),
                cfg.getUserPassword()
        );
    }
}
